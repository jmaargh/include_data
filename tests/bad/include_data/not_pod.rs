fn main() {}

static NOT_POD: char = include_data::include_data!("../../test_data/binary_4");

// This type satisfies the requirements of `Pod`, but is not marked as such
#[repr(transparent)]
struct Foo(u16);

static NOT_POD_CUSTOM: Foo = include_data::include_data!("../../test_data/binary_2");
