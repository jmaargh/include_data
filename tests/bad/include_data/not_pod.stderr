error[E0277]: the trait bound `char: Pod` is not satisfied
 --> tests/bad/include_data/not_pod.rs:3:24
  |
3 | static NOT_POD: char = include_data::include_data!("../../test_data/binary_4");
  |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |                        |
  |                        the trait `Pod` is not implemented for `char`
  |                        this tail expression is of type `char`
  |                        required by a bound introduced by this call
  |
  = help: the following other types implement trait `Pod`:
            ()
            ManuallyDrop<T>
            Option<T>
            PhantomData<T>
            PhantomPinned
            Wrapping<T>
            [T; N]
            __m128
          and $N others
note: required by a bound in `NOT_POD::typecheck`
 --> tests/bad/include_data/not_pod.rs:3:24
  |
3 | static NOT_POD: char = include_data::include_data!("../../test_data/binary_4");
  |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `typecheck`
  = note: this error originates in the macro `include_data::include_data` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Foo: Pod` is not satisfied
 --> tests/bad/include_data/not_pod.rs:9:30
  |
9 | static NOT_POD_CUSTOM: Foo = include_data::include_data!("../../test_data/binary_2");
  |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |                              |
  |                              the trait `Pod` is not implemented for `Foo`
  |                              this tail expression is of type `Foo`
  |                              required by a bound introduced by this call
  |
  = help: the following other types implement trait `Pod`:
            ()
            ManuallyDrop<T>
            Option<T>
            PhantomData<T>
            PhantomPinned
            Wrapping<T>
            [T; N]
            __m128
          and $N others
note: required by a bound in `NOT_POD_CUSTOM::typecheck`
 --> tests/bad/include_data/not_pod.rs:9:30
  |
9 | static NOT_POD_CUSTOM: Foo = include_data::include_data!("../../test_data/binary_2");
  |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `typecheck`
  = note: this error originates in the macro `include_data::include_data` (in Nightly builds, run with -Z macro-backtrace for more info)
